include/picolibrary/microchip/megaavr/spi.h

```diff
@@ -233,6 +234,177 @@ class Basic_Controller<Peripheral::SPI> {
+     */
+    Basic_Controller( GPIO::Push_Pull_IO_Pin xck, Peripheral::USART & usart ) noexcept :
+        m_xck{ std::move( xck ) },
+        m_usart{ &usart }
+    {
+        m_usart->disable();
+
+        m_usart->configure();
+    }
+
+    /**
+     * \brief Constructor.
+     *
+     * \param[in] source The source of the move.
+     */
+    constexpr Basic_Controller( Basic_Controller && source ) noexcept :
+        m_xck{ std::move( source.m_xck ) },
+        m_usart{ source.m_usart }
+    {
+        source.m_usart = nullptr;
+    }
+
+    Basic_Controller( Basic_Controller const & ) = delete;
+
+    /**
+     * \brief Destructor.
+     */
+    ~Basic_Controller() noexcept
+    {
+        disable();
+    }
+
+    /**
+     * \brief Assignment operator.
+     *
+     * \param[in] expression The expression to be assigned.
+     *
+     * \return The assigned to object.
+     */
+    auto & operator=( Basic_Controller && expression ) noexcept
+    {
+        if ( &expression != this ) {
+            disable();
+
+            m_xck   = std::move( expression.m_xck );
+            m_usart = expression.m_usart;
+
+            expression.m_usart = nullptr;
+        } // if
+
+        return *this;
+    }
+
+    auto operator=( Basic_Controller const & ) = delete;
+
+    /**
+     * \brief Initialize the controller's hardware.
+     *
+     * \return Success.
+     */
+    auto initialize() noexcept -> Result<Void, Void>
+    {
+        static_cast<void>( m_xck.initialize() );
+
+        m_usart->enable();
+
+        return {};
+    }
+
+    /**
+     * \brief Configure the controller's clock, and data exchange bit order to meet a
+     *        specific device's communication requirements.
+     *
+     * \param[in] configuration The clock, and data exchange bit order configuration that
+     *            meets the device's communication requirements.
+     *
+     * \return Success.
+     */
+    auto configure( Configuration const & configuration ) noexcept -> Result<Void, Void>
+    {
+        m_usart->configure(
+            configuration.clock_polarity,
+            configuration.clock_phase,
+            configuration.bit_order,
+            configuration.scaling_factor );
+
+        return {};
+    }
+
+    /**
+     * \brief Exchange data with a device.
+     *
+     * \param[in] data The data to transmit.
+     *
+     * \return The received data.
+     */
+    auto exchange( std::uint8_t data ) noexcept -> Result<std::uint8_t, Void>
+    {
+        while ( not m_usart->transmit_buffer_empty() ) {}
+
+        m_usart->transmit( data );
+
+        while ( not m_usart->data_available() ) {}
+
+        return m_usart->receive<std::uint8_t>();
+    }
+
+  private:
+    /**
+     * \brief The XCK pin.
+     */
+    GPIO::Push_Pull_IO_Pin m_xck{};
+
+    /**
+     * \brief The USART peripheral used by the SPI controller.
+     */
+    Peripheral::USART * m_usart{};
+
+    /**
+     * \brief Disable the SPI.
```

Comment:
This function disables the USART (peripheral and associated hardware interface), not the SPI (peripheral and associated hardware interface). Please update the comment accordingly.